### ISSUE TITLE
DatePicker can overflow when `initialEntryMode` is `DatePickerEntryMode.input` in landscape

### ISSUE THREAD

**Issue by MdSaLaMa** [flutter/flutter#132544]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

DatePicker dialog overflow when 
```dart
 initialEntryMode=DatePickerEntryMode.input,
```
in landscape orientation.

### Expected results

DatePicker dialog should be scrolling like in TimePicker when showing keyboard.

### Actual results

DatePicker dialog overflow when 
```dart
 initialEntryMode=DatePickerEntryMode.input,
```
in landscape orientation.

### Code sample

<details><summary>Code sample</summary>

```dart
 await showDatePicker(
                  context: context,
                  initialDate: dateValue ?? DateTime.now(),
                  firstDate: DateTime(2020),
                  lastDate: DateTime.now().add(Duration(days: 400)),
                  initialEntryMode: DatePickerEntryMode.input,
                  helpText: label,
                  fieldLabelText: label)
```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

![Screenshot_2023-08-15-06-54-07-239](https://github.com/flutter/flutter/assets/38154967/8f7f4c0a-f831-4e43-a0eb-8d57bde96748)
![Screenshot_2023-08-15-06-54-18-366](https://github.com/flutter/flutter/assets/38154967/6d8b8160-97f4-444d-977b-ded4a864ad25)

</details>

### Logs

<details><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 3.10.0, on Microsoft Windows [Version 10.0.19043.1526], locale en-US)

```

</details>

**Comment by danagbemava-nc** [comment]
Hi @MdSaLaMa, what device(s) are you seeing this on? You are also on an older version of flutter stable, can you upgrade to the latest version of flutter stable(3.10.6) to see if you still experience this issue?

**Comment by danagbemava-nc** [comment]
Hi @MdSaLaMa, what device(s) are you seeing this on? I tested it on my pixel 7 and it worked just fine on stable and the latest master. 
| master | stable |
| -- | -- |
| <img src="https://github.com/flutter/flutter/assets/88313112/a1d12b6c-1ed1-455e-aac6-3dc0a0d6708e" /> | <img src="https://github.com/flutter/flutter/assets/88313112/42f6ee5a-6683-414d-b364-62f81a82b065" />

**Comment by MdSaLaMa** [comment]
My device is Lenovo k6 (w=360p, h=640p), but as you see in screenshot above the keyboard has more height than in pixel 7 .
I think the solution is to wrap the DatePicker with SingleChildScrollView.
If you checked the TimePicker dialog you will find it scrollable in the same case.

**Comment by danagbemava-nc** [comment]
Yes, wrapping it in a Scrollable might be a good idea, we needed to know which device(s) it could reproduce on so that we could when it was fixed to see whether or not it was indeed fixed. 

Labeling based on the report shared above.

**Comment by HansMuller** [comment]
CC @TahaTesser

**Comment by TahaTesser** [comment]
Investigated this issue and compared it with Android components. I can reproduce this issue using a 1080p device but using a keyboard that shows a toolbar like Grammarly keyboard.

Here you can see there isn't enough room when the keyboard appears in the landscape mode. 

| Android Components | 
| --------------- | 
| <img src="https://github.com/flutter/flutter/assets/48603081/356d634c-89c1-41e6-99af-2cd854c11a60" />	|

| Futter | 
| --------------- | 
| <img src="https://github.com/flutter/flutter/assets/48603081/b6dc2fa4-0ba2-478c-b96b-13107c7b55e0" /> | 


1. The keyboard on top of the dialog in the Android components.
2. Dialog takes less spacing in the safe area region While Fluter has a lot bigger spacing.

![Group 1](https://github.com/flutter/flutter/assets/48603081/7dfd43fc-a5ff-4aec-904a-9b2f5724750f)
